### PR TITLE
Update dependencies and add overloads to allow custom SMTP Server Ports

### PR DESCRIPTION
**Pull request by dougkwilson**
See #33 for more.

Did not change any existing API, just added overloads to preserve backwards compatibility.

### CODE REVIEW

**Comment by nblumhardt** [comment]
Thanks for the PR, Doug 👍 

The backwards-compatible overload is sound, but over time if we use this approach we end up having to create a lot of potentially overlapping method signatures to cope with new options (i.e. each overload needs a new overload for each new feature).

There's an alternative documented in [this comment](https://github.com/serilog/serilog-sinks-elasticsearch/pull/150#issuecomment-364793144) that should work here - would it be possible to switch to that approach to add an `int port = 25` optional parameter?

Thanks again!

**Comment by dougkwilson** [comment]
I'll definitely take a look.

**Comment by dougkwilson** [comment]
All right... that took a lot more effort than I anticipated.  I followed the example suggested and I think I've got something that works that will also allow for the future.

But the more that I worked with this, the more that I think it needs a significant rewrite and version bump.

File: src/Serilog.Sinks.Email/LoggerConfigurationEmailExtensions.cs [hunk 1]
```diff
@@ -176,6 +166,7 @@ public static class LoggerConfigurationEmailExtensions
         /// <param name="mailSubject">The subject, can be a plain string or a template such as {Timestamp} [{Level}] occurred.</param>
         /// <returns>Logger configuration, allowing configuration to continue.</returns>
         /// <exception cref="ArgumentNullException">A required parameter is null.</exception>
+        [Obsolete("New code should not be compiled against this obsolete overload"), EditorBrowsable(EditorBrowsableState.Never)]
         public static LoggerConfiguration Email(
```
[the remark this answers is not in the document]
Comment (dougkwilson) [reply src/Serilog.Sinks.Email/LoggerConfigurationEmailExtensions.cs]: You can't configure EmailConnectionInfo from appsettings.json with Serilog.Settings.Configuration.  EmailConfigurationInfo completely defeats the point of having default values for the parameters.

**Comment by dougkwilson** [comment]
The more I work with this sink the more frustrated I become.  [This issue](https://github.com/serilog/serilog-sinks-email/issues/46#issue-263269398) is a complete showstopper for me as fixing risks breaking existing usages.  

The suggestion [here](https://github.com/serilog/serilog-sinks-email/issues/42#issuecomment-343315183) of splitting the sink has a lot of merit and I think is the correct way to go.  

I'm going to close this PR...

**Comment by nblumhardt** [comment]
Yep, know the feeling 👍

Logging directly to SMTP email is often a pretty awkward experience - email has too many points of failure of its own to be a good channel except in the most dire circumstances.

If you have access to a monitoring tool that can do email alerting with some kind of aggregation of events, plus rate limiting, that's the way to go.

**Pull request closed without merging**